suite/tests/linux/syscall_pwait.c

```diff
@@ -119,15 +137,18 @@ main(int argc, char *argv[])
         /* XXX i#3240: DR currently does not handle the atomicity aspect of this system
          * call. Once it does, please include this in this test or add a new test.
          */
-        sigset_t *test_set_check = &test_set;
+        sigset_t pre_syscall_set;
+        sigprocmask(SIG_SETMASK, NULL, &pre_syscall_set);
         if (epoll_pwait(epoll_fd, &events, 24, -1, &test_set) == -1) {
             if (errno != EINTR)
                 perror("expected EINTR");
-            if (&test_set != test_set_check)
-                perror("sigmask mismatch");
         } else {
             perror("expected interruption of syscall");
         }
+        sigset_t post_syscall_set;
+        sigprocmask(SIG_SETMASK, NULL, &post_syscall_set);
+        if (!compare_sigmasks(&pre_syscall_set, &post_syscall_set))
+            perror("sigmask mismatch");
```

Comment:
perror acts on errno, but your function compare_sigmasks does not set errno

Reply:
It still prints the message though and fails. I change it into print